src/components/Form/Checkbox.tsx

```diff
@@ -0,0 +1,175 @@
+   */
+  variant?: CheckboxVariant;
+  /**
+   * 'default' | 'error' | 'disabled'
+   * @default default
+   */
+  status?: CheckboxStatus;
+  /**
+   * Status text to be displayed in the status text element.
+   */
+  statusText?: string;
+  /**
+   * Hint text to be displayed under the label.
+   */
+  hintText?: string;
+  /** Pass custom props to Checkbox's input component/element */
+  checkboxInputProps?: CheckboxInputProps;
+  /**
+   * aria-label for the HTML input-element,
+   * alternatively you can define aria-labelledby with label-element id
+   */
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+  /** Unique id
+   * @default uuidV4
+   */
+  id?: string;
+}
+
+export interface CheckboxState {
+  checkboxState: boolean;
+}
+
+export class Checkbox extends Component<CheckboxProps> {
+  state = {
+    checkboxState: !!this.props.checked || !!this.props.defaultChecked || false,
+  };
+
+  /**
+   * This is deprecated and should be replaced
+   */
+  componentWillReceiveProps(nextProps: CheckboxProps) {
+    const { checked } = nextProps;
+    if (!!checked) {
+      this.setState({ checkboxState: !!checked });
+    }
+  }
+
+  handleClick = () => {
+    const { checked, onClick } = this.props;
+    const { checkboxState } = this.state;
+    if (checked === undefined) {
+      this.setState({ checkboxState: !checkboxState });
+    }
+    if (!!onClick) {
+      onClick({ checkboxState: !checkboxState });
+    }
+  };
+
+  render() {
+    const {
+      id: propId,
+      className,
+      disabled = false,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledBy,
+      children,
+      checkboxInputProps,
+      checked: dismissChecked,
+      defaultChecked: dismissDefaultChecked,
+      onClick: dismissOnClick,
+      hintText,
+      status,
+      statusText,
+      ...passProps
+    } = this.props;
+    const { checkboxState } = this.state;
+
+    const id = idGenerator(propId);
+    const statusTextId = `${idGenerator(propId)}-statusText`;
+    const hintTextId = `${idGenerator(propId)}-hintText`;
+
+    const newCheckboxInputProps = {
+      disabled,
+      id,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledBy,
+      checked: !!checkboxState,
+      className: checkboxBaseClassNames.input,
+      onChange: this.handleClick,
+      ...checkboxInputProps,
+    };
+
+    return (
+      <HtmlDiv
+        className={classnames(checkboxBaseClassNames.container, className, {
+          [checkboxBaseClassNames.disabled]: !!disabled,
+        })}
+      >
+        <HtmlInput {...newCheckboxInputProps} type="checkbox" />
+        <HtmlLabel
+          htmlFor={id}
+          className={classnames(className, checkboxBaseClassNames.label)}
+          {...passProps}
+        >
+          {children}
+        </HtmlLabel>
+        {hintText ? (
+          <HtmlSpan className={checkboxBaseClassNames.hintText} id={hintTextId}>
+            {hintText}
+          </HtmlSpan>
+        ) : null}
```

Comment:
This might be more readable without the need of ternary operator and returning `null` value.

After change it would be like:

```tsx
{!!hintText && (
	<HtmlSpan className={checkboxBaseClassNames.hintText} id={hintTextId}>
		{hintText}
	</HtmlSpan>
)}
```

Reply:
Forgot about that syntax. Thanks :)

Reply:
Replaced with the truthy syntax.